docs/content/middlewares/http/ratelimit.md

```diff
@@ -496,3 +496,783 @@ http:
     [http.middlewares.test-ratelimit.rateLimit.sourceCriterion]
       requestHost = true
 ```
+
+### `redis`
+
+Using `redis` to store the tokens in token bucket algorithm. If none are set, the default is to use the in-memory of traefik to store the tokens
+
+Same usage as [this](https://doc.traefik.io/traefik/providers/redis)
+
+#### `redis.endpoints`
+
+_Required, Default="127.0.0.1:6379"_
+
+Defines how to access Redis.
+
+```yaml tab="Docker & Swarm"
+labels:
+    - "traefik.http.middlewares.test-ratelimit.ratelimit.burst=100"
+    - "traefik.http.middlewares.test-ratelimit.ratelimit.redis.endpoints=127.0.0.1:6379"
+```
+
+```yaml tab="Kubernetes"
+apiVersion: traefik.io/v1alpha1
+kind: Middleware
+metadata:
+    name: test-ratelimit
+spec:
+    rateLimit:
+        burst: 100
+        redis:
+            endpoints:
+                - "127.0.0.1:6379"
+```
+
+```yaml tab="Consul Catalog"
+- "traefik.http.middlewares.test-ratelimit.ratelimit.burst=100"
+- "traefik.http.middlewares.test-ratelimit.ratelimit.redis.endpoints=127.0.0.1:6379"
+```
+
+```yaml tab="File (YAML)"
+http:
+    middlewares:
+        test-ratelimit:
+            rateLimit:
+                burst: 100
+                redis:
+                    endpoints:
+                        - "127.0.0.1:6379"
+```
+
+```toml tab="File (TOML)"
+[http.middlewares]
+  [http.middlewares.test-ratelimit.rateLimit]
+    burst = 100
+    [http.middlewares.test-ratelimit.rateLimit.redis]
+      endpoints = ["127.0.0.1:6379"]
+```
+
+#### `redis.username`
+
+_Optional, Default=""_
+
+`username` is the username for Sentinel authentication.
```

Comment:
The `username` is used on the `UniversalClient` but no `MasterName` are configured. We are therefore not going to end up with a `FailoverClient` that support authenticating on a Sentinel cluster. If a Sentinel node is provided on the entrypoints it will be used as a standard redis server. So I think it would be wrong to talk about Sentinel here.
Though, that's something we could add in a follow up PR.

```suggestion
`username` is the username used to authenticate with the Redis server.
```